code-challenges.test.js

```diff
@@ -14,33 +14,137 @@
+
+// Function
+
+const nameJob = (array) => {
+	return array.map(
+		({ name, occupation }) =>
+			name
+				.split(" ")
+				.map((string) => string.charAt(0).toUpperCase() + string.slice(1))
+				.join(" ") +
+			" is " +
+			occupation +
+			"."
+	)
+}
+
+// PASSED
+
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("numsDiv3", () => {
+	it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+		const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+		// Expected output: [ 2, 0, -1, 0 ]
+		const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+		// Expected output: [ 2, 1, -1 ]
+		expect(numsDiv3(hodgepodge1)).toEqual([2, 0, -1, 0])
+		expect(numsDiv3(hodgepodge2)).toEqual([2, 1, -1])
+	})
+})
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+// FAILED:   ReferenceError: numsDiv3 is not defined
 
 // b) Create the function that makes the test pass.
 
+// Pseudo
+
+// Input: array
+// Output: array of only the remainders of the numbers divided by 3
+// Expected: [ 2, 0, -1, 0 ], [ 2, 1, -1 ]
+
+// Process
+
+// Declare a function called numDiv3 with a parameter of array
+// Declare a variable called numbers that is equal to array.filter value === number
+// Return numbers and map through each value with modulo 3
+// This should return all the remainders of each number divided by 3
+
+// Function:
+
+const numsDiv3 = (array) => {
+	const numbers = array.filter((value) => typeof value === "number")
+	return numbers.map((value) => value % 3)
+}
+
+// PASSED
+
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe("numSum", () => {
+	it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+		const cubeAndSum1 = [2, 3, 4]
+		// Expected output: 99
+		const cubeAndSum2 = [0, 5, 10]
+		// Expected output: 1125
+		expect(numSum(cubeAndSum1)).toEqual(99)
+		expect(numSum(cubeAndSum2)).toEqual(1125)
+	})
+})
+
+// FAILED: ReferenceError: numSum is not defined
 
 // b) Create the function that makes the test pass.
+
+// Pseudo
+
+// Input: array
+// Output: sum of values ** 3
+// Expected: 99, 1125
+
+// Process
+
+// Declare the function numSum with a parameter of array
+// Declare a variable newArr
+// Set the newArr to array.map so it takes the index and cubes each value (** 3)
+// Return newArr.reduce to add a + b
```

Comment:
Good flow of logic, and nice use of `.reduce`!
- The pseudo is a little lacking in the explanation of how `.reduce` is working, wouldn't hurt to elaborate a little more on that